### ISSUE TITLE
allowLocal should support export statements

### ISSUE THREAD

**Issue by cartant** [cartant/eslint-plugin-etc#21]
The `allowLocal` option in `no-const-enum` and `prefer-interface` should support export statements. For example:

```ts
type Person = { name: string; };
export { Person };
```

Also, with `prefer-interface`, `allowLocal` should deem indirectly exported types to be non-local, as the whole point of the rule is to avoid inlining.
